docs/articles/documentation/concepts/browsers.md

```diff
@@ -0,0 +1,301 @@
+    .browsers('firefox:path/to/firefox:headless')
+    .run();
+```
+
+Note that the [Take screenshot](../guides/basic-guides/interact-with-the-page.md#take-screenshot) and [Resize window](../guides/basic-guides/interact-with-the-page.md#resize-window) features are fully supported in the headless mode.
+
+### Automation Port
+
+Chrome and Firefox require a remote control port for TestCafe to attach to the browser instance. TestCafe automatically assigns a free port but you can specify a custom port via the `cdpPort` argument for Chrome and the `marionettePort` argument for Firefox.
+
+```sh
+testcafe "chrome:headless:cdpPort=9223" tests/sample-fixture.js
+```
+
+```js
+runner
+    .src('tests/sample-fixture.js')
+    .browsers('chrome:headless:cdpPort=9223')
+    .run();
+```
+
+```sh
+testcafe "firefox:headless:marionettePort=9223" tests/sample-fixture.js
+```
+
+```js
+runner
+    .src('tests/sample-fixture.js')
+    .browsers('firefox:headless:marionettePort=9223')
+    .run();
+```
+
+## Use Chromium Device Emulation
+
+You can run test with Chromium's built-in [device emulator](https://developers.google.com/web/tools/chrome-devtools/device-mode/) in Google Chrome, Chromium and Chromium-based Microsoft Edge. To do this, use the `emulation` browser parameter.
+
+### Emulate a Device
+
+Specify the target device with the `device` parameter.
+
+```sh
+testcafe "chrome:emulation:device=iphone X" tests/sample-fixture.js
+```
+
+```js
+runner
+    .src('tests/sample-fixture.js')
+    .browsers('chrome:emulation:device=iphone X')
+    .run();
+```
+
+### Emulate a Screen Size
+
+You can specify parameters like `width`, `height`, or `orientation` to configure the device emulator.
+
+```sh
+testcafe "chrome:emulation:width=100;height=200;mobile=true;orientation=vertical;touch=true" tests/sample-fixture.js
+```
+
+```js
+runner
+    .src('tests/sample-fixture.js')
+    .browsers('chrome:emulation:width=100;height=200;mobile=true;orientation=vertical;touch=true')
+    .run();
+```
+
+### Use Emulation in Headless Mode
+
+You can combine both device emulation and headless mode.
+
+```sh
+testcafe "chrome:headless:emulation:device=iphone X;cdpPort=9223" tests/sample-fixture.js
+```
+
+```js
+runner
+    .src('tests/sample-fixture.js')
+    .browsers('chrome:headless:emulation:device=iphone X;cdpPort=9223')
+    .run();
+```
+
+### Use Emulation in Portable Chrome
+
+To enable device emulation in a portable Chrome, use the [browser alias](#locally-installed-browsers). The `path:` prefix does not work in this case.
+
+```sh
+testcafe "chrome:d:\chrome_portable\chrome.exe:emulation:device=iphone X" tests/test.js
+```
+
+### Substitute a User Agent
+
+Use the `userAgent` parameter to substitute a user agent string.
+
+> Important! TestCafe relies on the user agent string to emulate the browser behavior. Tests are not guaranteed to run correctly if you provide a user agent that is invalid or [not supported by TestCafe](#browser-support).
```

Comment:
Change "emulate the browser behavior" to "emulate browser behavior"

In the second sentence, can you be more specific than "provide"?

Reply:
Can we change it to:
Note: TestCafe relies on a user agent string to emulate browser behavior. Tests are not guaranteed to run correctly if you specify a user agent that is invalid or [not supported by TestCafe](#browser-support).

Reply:
Yes, sure, I did it.

BTW, `Important!` is a part of our custom markdown denoting that this is an 'Important!' box. The word itself is displayed only when you view doc's sources on GitHub.

Reply:
Okay, please make my change above with "Important!" instead of "Note:"